app/controllers/folders_controller.rb

```diff
@@ -11,9 +12,13 @@ def create
     @folder = Folder.new(folder_params)
     if @folder.save
       flash[:success] = "#{@folder.name} Successfully Created"
-      redirect_to home_path
+      if current_folder
```

Comment:
@alfosco I think we can take out this second if/else statement because we're storing the session even if it's the root folder.